PORT-6857 Mini guide to create ECR image with tags (#959)
port-labs/port-docs: docs/create-self-service-experiences/setup-backend/github-workflow/examples/aws/push-image-to-ecr.md

import PortTooltip from "/src/components/tooltip/tooltip.jsx";

# Generate ECR image with tags

> In the following guide, you are going to create a [self-service action](https://docs.getport.io/create-self-service-experiences/) in Port that executes a [GitHub workflow](https://docs.getport.io/create-self-service-experiences/setup-backend/github-workflow) to:

Reviewer: ```suggestion
In the following guide, we are going to create a [self-service action](https://docs.getport.io/create-self-service-experiences/) in Port that executes a [GitHub workflow](https://docs.getport.io/create-self-service-experiences/setup-backend/github-workflow) to:
```

Reviewer: Use internal and not full URLs (meaning that when the user clicks on the link, it does not open a new tab for him)

Author: alright

> - Generate a Docker image with tags and,

Reviewer: ```suggestion
- Generate a Docker image with tags
```

> - Push it to an [AWS Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/).

Reviewer: ```suggestion
- Push the image to an [AWS Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/).
```

<details>

<summary>GitHub Workflow</summary>

```yaml showLineNumbers title="create-and-push-image.yml"
name: Create and Push Image to ECR

on:
  workflow_dispatch:
    inputs:
      image_repo:
        description: 'Repository URL'
        required: true
      dockerfile:
        description: 'Path to Dockerfile'
        required: true
      port_payload:
        required: true
        description: "Port's payload, including details for who triggered the action and general context (blueprint, run id, etc...)"
        type: string

jobs:
  build:
    runs-on: ubuntu-latest

    env:
      REPO_URL: ${{ fromJson(inputs.port_payload).payload.entity.properties.url }}
      TRIGGERED_BY: ${{ fromJson(inputs.port_payload).trigger.by.user.email || github.actor }}

    steps:
    - name: Extract repository owner and name
      run: |
        repo_owner=$(echo "${REPO_URL}" | awk -F/ '{print $4}')
        repo_name=$(echo "${REPO_URL}" | awk -F/ '{print $5}')

        echo "REPO_OWNER=$repo_owner" >> $GITHUB_ENV
        echo "REPO_NAME=$repo_name" >> $GITHUB_ENV
      shell: bash
  
    - name: Checkout repository

>       uses: actions/checkout@v2

Reviewer: ```suggestion
      uses: actions/checkout@v4
```
